File: AppSets-Android/share/src/main/java/xcj/app/share/http/model/ContentInfo.kt
@file:OptIn(ExperimentalEncodingApi::class)

package xcj.app.share.http.model

import kotlin.io.encoding.Base64
import kotlin.io.encoding.ExperimentalEncodingApi

/**
 * @param id is DataContent.id
 * @param type, 0:String, 1:File
 */
data class ContentInfo(
    val id: String,
    var name: String,
    val size: Long = 0,
    val type: Int = 0,
) {
    companion object{
        const val TYPE_STRING = 0
        const val TYPE_FILE = 1
        const val TYPE_URI = 2
        const val TYPE_BYTES = 3
    }
    fun decode(): ContentInfo {
        val decodeName = Base64.decode(name).decodeToString()
        name = decodeName
        return this
    }

    fun encode(): ContentInfo {
        val encodeName = Base64.encode(name.toByteArray())
        name = encodeName
        return this
    }
}

File: AppSets-iOS/AppSets/com/app/appsets/share/http/model/ContentInfo.swift
//
//  ContentInfo.swift
//  AppSets
//
//  Created by caiju Xu on 2025/3/3.
//
import Foundation

class ContentInfo: Codable {
    public static let TYPE_STRING = 0
    public static let TYPE_FILE = 1
    public static let TYPE_URI = 2
    public static let TYPE_BYTES = 3

    var id: String
    var name: String
    var size: CLong
    var type: Int
    
    init(id: String, name: String, size: CLong, type: Int) {
        self.id = id
        self.name = name
        self.size = size
        self.type = type
    }
    
    enum CodingKeys: CodingKey {
        case id
        case name
        case size
        case type
    }

    func encode() -> ContentInfo {
        guard let nameData = name.data(using: .utf8) else {
            return self
        }
        name = nameData.base64EncodedString()
        return self
    }

    func decode() -> ContentInfo {
        guard let nameData = Data.init(base64Encoded: name) else {
            return self
        }
        guard let decodeName = String(data: nameData, encoding: .utf8) else {
            return self
        }
        name = decodeName

        return self
    }
}
